superset/db_engine_specs/base.py

```diff
@@ -387,6 +387,19 @@ def df_to_sql(cls, df: pd.DataFrame, **kwargs):  # pylint: disable=invalid-name
         """
         df.to_sql(**kwargs)
 
+    @classmethod
+    def dialect_paramstyle(cls, sqla_uri: str) -> Optional[str]:
+        """
+        Specify a paramstyle for certain dialects. Using the default param
+        style for certain drivers causes issues when special characters, eg (,(,%
+        are used in values.
+        """
+        dialect = ""
+        matches = re.search(r"^([^:+]+)[:+]", sqla_uri, re.IGNORECASE)
```

Comment:
SQLALchemy has a [`make_url`](https://docs.sqlalchemy.org/en/13/core/engines.html#sqlalchemy.engine.url.make_url) method which returns a URL object from which the dialect can be obtained via `get_dialect()`. No regex required.